planner/core/exhaust_physical_plans.go

```diff
@@ -950,6 +951,10 @@ func (ijHelper *indexJoinBuildHelper) buildTemplateRange(matchedKeyCnt int, eqAn
 				if err != nil {
 					return nil, err
 				}
+				ran.LowVal[i], err = ran.LowVal[i].ConvertTo(sc, sf.GetArgs()[0].GetType())
```

Comment:
Should we consider the overflow error or if the value is null?

Reply:
Reasonable, i'll add test to cover this case.

Reply:
Add one case.
And
```
// ConvertTo converts a datum to the target field type.
func (d *Datum) ConvertTo(sc *stmtctx.StatementContext, target *FieldType) (Datum, error) {
	if d.k == KindNull {
		return Datum{}, nil
	}
```
NULL is safe currently.

Reply:
What about overflow?

Reply:
Can we put the check of `types.ErrOverflow` into `ConvertTo`? since there are a lot of codes checking this after returning from `ConvertTo`.

Reply:
Emm, the `convertPoint` in `ranger` didn't consider the overflow error too. Let me check it.